subworkflows/local/minimap2_subworkflow.nf

```diff
@@ -19,8 +19,15 @@ workflow MINIMAP2_SUBWORKFLOW {
     ch_versions = ch_versions.mix(MINIMAP2_INDEX.out.versions)
     ch_index = MINIMAP2_INDEX.out.index
 
+    // match index to the corresponding reads
```

Comment:
Wait isn't this accomplished using either `cross` or `join` (depending on the output channel structure you want)?

Here's my understanding of the problem: 
* You have an input channel `reads` of the format `[meta, path(fasta)]` and another channel `assembly` of the format `[meta, path(fasta)]`. 
* When the `assembly` channel goes into the `MINIMAP2_INDEX`, the order of the output in terms of `meta` changes, bc Nextflow processes them in a different order.
* And you want the inputs from `reads` and the output of `MINIMAP2_INDEX` to have the same sample order so they can processed. This is accomplished by the `meta` field, which is the sample ID. 
* Is this accurate?

I think you can do `ch_mapping = reads.join(ch_index)` and that should work. Then you'd have the following channel structure: `[[meta, path(fasta), path(index)]]`. You'd have to adjust the input of the `MINIMAP2_ALIGN` obviously.

Or you can do `cross` which would give you: a channel: `[[meta, path(fasta)], [meta, path(index)]]`.

I am sure it could require some fiddling, but this should be so much cleaner than what you have here.

Reply:
Ah I didn't realize this is what `join` was doing. I think I tried `cross` and it didn't quite work like I wanted but can try again.

Reply:
Let me know once you try, we can also try to debug together! One of those should definitely work.